bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,570 @@
+    return endpoint.substring(0, endpoint.lastIndexOf(":"));
+  }
+
+  @Override
+  public String getAdminHost() {
+    String endpoint = tableAdminSettings.getStubSettings().getEndpoint();
+    return endpoint.substring(0, endpoint.lastIndexOf(":"));
+  }
+
+  @Override
+  public int getPort() {
+    String endpoint = dataSettings.getStubSettings().getEndpoint();
+    return Integer.parseInt(endpoint.substring(endpoint.lastIndexOf(":") + 1));
+  }
+
+  @Override
+  public int getBulkMaxRowCount() {
+    return dataSettings
+        .getStubSettings()
+        .bulkMutateRowsSettings()
+        .getBatchingSettings()
+        .getElementCountThreshold()
+        .intValue();
+  }
+
+  // ************** Getters **************
+  public boolean isChannelPoolCachingEnabled() {
+    // This is primarily used by Dataflow where connections open and close often. This is a
+    // performance optimization that will reduce the cost to open connections.
+    return configuration.getBoolean(BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL, false);
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableDataSettings}. */
+  public BigtableDataSettings getDataSettings() {
+    return dataSettings;
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableTableAdminSettings}. */
+  public BigtableTableAdminSettings getTableAdminSettings() {
+    return tableAdminSettings;
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableInstanceAdminSettings}. */
+  public BigtableInstanceAdminSettings getInstanceAdminSettings() {
+    if (!isNullOrEmpty(configuration.get(BIGTABLE_EMULATOR_HOST_KEY))) {
+      return null;
+    }
+    return instanceAdminSettings;
+  }
+
+  // ************** Private Helpers **************
+  private BigtableDataSettings.Builder buildBigtableDataSettings() throws IOException {
+    BigtableDataSettings.Builder dataBuilder =
+        BigtableDataSettings.newBuilder()
+            .setProjectId(getProjectId())
+            .setInstanceId(getInstanceId());
+
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+    if (!isNullOrEmpty(appProfileId)) {
+      dataBuilder.setAppProfileId(appProfileId);
+    }
+
+    EnhancedBigtableStubSettings.Builder stubSettings = dataBuilder.stubSettings();
+
+    buildEndpoints(stubSettings, BIGTABLE_HOST_KEY);
+
+    buildHeaderProvider(stubSettings);
+
+    buildCredentialProvider(stubSettings);
+
+    String shortRpcTimeoutStr = configuration.get(BIGTABLE_RPC_TIMEOUT_MS_KEY);
+    if (!isNullOrEmpty(shortRpcTimeoutStr)) {
+      // rpcTimeout & totalTimeout for non-retry operations.
+      Duration shortRpcTimeout = ofMillis(Long.parseLong(shortRpcTimeoutStr));
+
+      stubSettings.checkAndMutateRowSettings().setSimpleTimeoutNoRetries(shortRpcTimeout);
+
+      stubSettings.readModifyWriteRowSettings().setSimpleTimeoutNoRetries(shortRpcTimeout);
+    }
+
+    buildBulkMutationsSettings(stubSettings);
```

Comment:
For the per method RPC settings I think it will be a lot cleaner to structure it like:

buildBulkMutationsSettings(stubSettings.bulkMutationSettings())
buildBulkReadRowsSettings(stubSettings.bulkReadRowSettings())
...
buildIdempotentCallSettings(stubSettings.mutateRowSettings())
buildIdempotentCallSettings(stubSettings. readRowSettings())
...
buildNonIdempotentCallSettings(stubSettings.checkAndMutateRowSettings())
...
void buildIdempotentUnaryCallSettings(UnaryCallSettings<?,?> callSettings) {
  // Configure the callSettings using INITIAL_ELAPSED_BACKOFF_MILLIS_KEY, BIGTABLE_USE_TIMEOUTS_KEY, ENABLE_GRPC_RETRIES_KEY, ADDITIONAL_RETRY_CODES, etc
}